docs/context.md

```diff
@@ -0,0 +1,36 @@
+---
+title: 'Context concept'
+weight: 50
+---
+
+## Overview
+
+AsyncAPI CLI provides functionality called `context`. It's purpose is to help to work with AsyncAPI CLI in large projects where you do not have just one service exposing AsyncAPI document, but multiple.
+
+Event driven architecture involves multiple actors, subscribers and publishers. One time you want to validate document **A** and the other time you want to generate models from document **B**. Every time you do it, you need to provide to AsyncAPI CLI the location of the AsyncAPI document, which might be time consuming. You can workaround it with aliases in bash profiles or with other solutions but it is better to use `context` feature as you can then store it in your repository and share with other team members.
+
+In short it means that for example instead of writing `asyncapi validate /some/folder/my-asyncapi.yml` you can create a context called `myasync` that will point to `/some/folder/my-asyncapi.yml`. This way next time you use the CLI you can do `asyncapi validate myasync`.
+
+### How to add context to a project
+
+Context can be added to a project in two ways:
+- using CLI, by executing command `asyncapi config context add [CONTEXT-NAME] [SPEC-FILE-PATH]`,
+- manually, by creation of file `.asyncapi-cli` of the predefined format anywhere in the repository up the path of main executable, or in a user's home directory.
```

Comment:
we need to clearly specify this fact that by default context is in home directory and that you can also have it in the project

```suggestion
### Context file location

You can have a global context for your workstation, and a project specific context.

If your use case is that you work with multiple repositories, you might want to use a global context. The `.asyncapi-cli` context file is then located in your home directory. This file is automatically created by the CLI once you run `asyncapi config context add` in a project that doesn't have its own context file

You can also store your custom `.asyncapi-cli` file in your project with custom configuration. This way when you run `asyncapi config context add` inside your project, the new context is added to the context file under your project.

### How to add context to a project

1. Create file `.asyncapi-cli` in the root of your project
2. Execute command `asyncapi config context add [CONTEXT-NAME] [SPEC-FILE-PATH]`
```

> Create file `.asyncapi-cli` in the root of your project

can it be empty file 

> Create file `.asyncapi-cli` in the root of your project

nice to have, but please not now 😄 to have `asyncapi config context init` that by default creates empty, valid file in project root, but if you do `asyncapi config context init ./mydir` it will create it in some other dir

Reply:
just enable that it can be an empty file, otherwise it will be difficult to really create a context from scratch. Enable that empty file is ok, and then it will enable people to add contexts with CLI and not manually by hand

Reply:
and you can open a separate issue, followup to create `init` command

Reply:
> nice to have, but please not now

The `asyncapi config context init` functionality relates to the code I had written hardly and intertwines with the doc, so I'll have to do it now.

Reply:
not sure about `[CONTEXT-NAME] [SPEC-FILE-PATH]` as this is what `add` is for. Also would be good to support `asyncapi config context init` to take currend dir by default